text/0750-feature-flag-advisor.md

```diff
@@ -0,0 +1,343 @@
+
+```
+> cdk flags "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021"
+    Description: Enable this feature flag to have cloudfront distributions use the security policy TLSv1.2_2021 by default.
+    Recommended Value: true
+    User Value: true
+```
+
+Additionally, to modify the state, you can run `cdk flags --set --flagName="#FLAGNAME#"`. This will return a list of values for you to choose from.
+
+```
+> cdk flags "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021" --set
+  What value would you like to change "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021" to?
+  > true
+    false
+> true
+
+Resynthesizing... 
+
+Here is the difference:
+- S3 Bucket:
+-    Properties:
+-      `BucketEncryption``:`` ``None`
++ S3 Bucket:
++    Properties:
++      BucketEncryption: 
++        `ServerSideEncryptionConfiguration``:`
++          ...
+```
+
+#### I want to revert my changes
+
+After performing any of the above operations, you will be shown the difference in CloudFormation templates and asked if you would like to
+keep the changes as shown below. If you choose to revert the changes, you are given options to revert all flags to its previous states
+or just a specific one. There will be a list of flags for you to choose from to revert.
+
+````
+Would you like to accept these changes? (Accept/revert)
+> revert
+Would you like to revert the entire feature flag configuration to its previous state or just revert the specific change(s)?
+[1] Revert all changes -> Restore all flags to their previous state
+[2] Revert only changes to one feature flag -> Undo a specific flag change
+[3] Exit without making any changes -> Keep your current configuration as is
+> 2
+Which feature flag would you like to revert?
+[1] @aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021
+[2] @aws-cdk/aws-s3:BucketEncryption
+[3] Exit
+> 1
+Reverting the change for @aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021...
+Resynthesizing...
+The flag has been reverted to its previous state.
+````
+
+## What are we trying to solve?
+
+Feature flags are an important tool for adding new bug fixes without affecting the current state of the application. However,
+feature flags often go unnoticed by the user. In order to change their configurations, users are required to manually update their
+`cdk.json` file, resulting in a time-consuming task. They can impact a user in many ways. For example, the
+@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021 flag changed the default behavior to use the updated version TLSv1.2_2021,
+which is a more secure and support TLS version for CloudFront. Users who stuck with the previous version may have been exposed to
+security vulnerabilities and failed compliance tests. To remain Backwards Compatible, newly released feature flags are automatically
+disabled for existing projects, meaning that those users are required to constantly keep an eye on CDK releases to manually update their feature flags.
```

Comment:
> keep an eye on CDK releases

A question comes to mind: Do we have a place that even lists all possible features? is it easy to get to?